### PR TITLE
added_orders_and_orderItems

### PR DESCRIPTION
**Pull request by SosunovychB**

### CODE REVIEW

File: src/main/java/book/store/controller/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            description = "Update order status")
+    public OrderDto updateOrderStatus(@PathVariable Long orderId,
+            @RequestBody @Valid UpdateOrderStatusRequestDto requestDto) {
+        return orderService.updateOrderStatus(orderId, requestDto);
+    }
+
+    private User getAuthenticatedUser() {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        return (User) authentication.getPrincipal();
+    }
```
Comment (koval-kn) [inline src/main/java/book/store/controller/OrderController.java]: it's a good practice to interact with context as less as possible
```suggestion
```

File: src/main/java/book/store/controller/OrderController.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    private final OrderItemService orderItemService;
+    private final OrderService orderService;
+
+    @PreAuthorize("hasRole('USER')")
+    @GetMapping
+    @Operation(summary = "Retrieve user's order history",
+            description = "Retrieve user's order history")
+    public List<OrderDto> getAllOrders(Pageable pageable) {
+        Long userId = getAuthenticatedUser().getId();
+        return orderService.getAllOrders(pageable, userId);
```
Comment (koval-kn) [inline src/main/java/book/store/controller/OrderController.java]: refactor other method accordingly
```suggestion
    public List<OrderDto> getAllOrders(Pageable pageable, Authentication authentication) {
        User user = (User) authentication.getPrincipal();
        return orderService.getAllOrders(pageable, user.getId());
```

File: src/main/java/book/store/repository/order/OrderRepository.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package book.store.repository.order;
+
+import book.store.model.Order;
+import java.util.List;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+public interface OrderRepository extends JpaRepository<Order, Long> {
+    @Query("FROM Order o LEFT JOIN FETCH o.orderItems i WHERE o.user.id = :userId")
```
Comment (koval-kn) [inline src/main/java/book/store/repository/order/OrderRepository.java:10]: it should work without custom queries
```suggestion
```

**Review by koval-kn — CHANGES_REQUESTED** [review]

File: src/main/java/book/store/repository/order/OrderRepository.java (hunk 3 above)
Comment (SosunovychB) [reply src/main/java/book/store/repository/order/OrderRepository.java:10]: Screenshot 2024-02-09 at 19 21 45
Unfortunately, it doesn't work.

Comment (koval-kn) [reply src/main/java/book/store/repository/order/OrderRepository.java:10]: so it should be fixed by yourself. see my suggestion

File: src/main/java/book/store/model/Order.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    @Enumerated(value = EnumType.STRING)
+    private Status status;
+    @Column(nullable = false)
+    private BigDecimal total;
+    @Column(name = "order_date", nullable = false)
+    private LocalDateTime orderDate;
+    @Column(name = "shipping_address", nullable = false)
+    private String shippingAddress;
+    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL)
+    private Set<OrderItem> orderItems = new LinkedHashSet<>();
```
Comment (koval-kn) [inline src/main/java/book/store/model/Order.java:47]:
```suggestion
  @ToString.Exclude
  @EqualsAndHashCode.Exclude
    private Set<OrderItem> orderItems = new LinkedHashSet<>();
```

**Review by koval-kn — CHANGES_REQUESTED** [review]

File: src/main/java/book/store/mapper/OrderItemMapper.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+public interface OrderItemMapper {
+    @Mapping(source = "book.id", target = "bookId")
+    OrderItemDto toDto(OrderItem orderItem);
+
+    @Named(value = "toOrderItemDtoSet")
+    default Set<OrderItemDto> toOrderItemDtoSet(Set<OrderItem> orderItems) {
+        return orderItems.stream()
+                .map(this::toDto)
+                .collect(Collectors.toSet());
+    }
```
Comment (koval-kn) [inline src/main/java/book/store/mapper/OrderItemMapper.java:22]:
```suggestion
    @IterableMapping(qualifiedByName = "toDto")
  Set<OrderItemDto> toDtoSet(Set<OrderItem> orderItems);
```

**Review by koval-kn — CHANGES_REQUESTED** [review]

File: src/main/java/book/store/mapper/OrderItemMapper.java (hunk 5 above)
Comment (SosunovychB) [reply src/main/java/book/store/mapper/OrderItemMapper.java:22]: @IterableMapping with custom query works, but @IterableMapping without custom query doesn't (still LazyIE).

**Pull request merged**